src/layers.cpp

```diff
@@ -115,12 +115,15 @@ Layer_::updateActiveLayers(void) {
 }
 
 void Layer_::updateHighestLayer(void) {
-  for (int8_t i = 31; i >= 0; i--) {
+  // If layer_count is set, start there, otherwise search from the
+  // highest possible layer for the top active layer
+  for (int8_t i = (layer_count - 1); i > 0; i++) {
```

Comment:
I...bet that logic's not quite what you meant ;)

Reply:
Heh. Well, at least it was only an 8-bit, signed integer, so it wouldn't take very long to become negative. Oops!